src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.test.js

```diff
@@ -56,4 +58,25 @@ describe("CSOAccount Component", () => {
 
     expect(setCsoAccountStatus).toHaveBeenCalled();
   });
+
+  test("On failed account creation, should redirect to parent application", async () => {
+    mock.onPost(API_REQUEST).reply(400, { message: "There was a problem." });
+
+    const { container } = render(
+      <CSOAccount
+        confirmationPopup={confirmationPopup}
+        applicantInfo={applicantInfo}
+        setCsoAccountStatus={setCsoAccountStatus}
+      />
+    );
+
+    expect(getByText(container, "Create CSO Account").disabled).toBeTruthy();
+    fireEvent.click(getByRole(container, "checkbox"));
+    expect(getByText(container, "Create CSO Account").disabled).toBeFalsy();
+
+    fireEvent.click(getByText(container, "Create CSO Account"));
+    await wait(() => {});
+
+    expect(window.open).toHaveBeenCalled();
```

Comment:
can we assert `toHaveBeenCalledWith("url", "_self")`? its a stronger assertion

Reply:
fixed